src/Umbraco.Infrastructure/PropertyEditors/MediaPicker3PropertyEditor.cs

```diff
@@ -126,23 +123,29 @@ public override object ToEditor(IProperty property, string? culture = null, stri
 
         public override object? FromEditor(ContentPropertyData editorValue, object? currentValue)
         {
-            if (editorValue.Value is JArray dtos)
+            // FIXME: consider creating an object deserialization method on IJsonSerializer instead of relying on deserializing serialized JSON here (and likely other places as well)
+            if (editorValue.Value is not JsonArray jsonArray)
             {
-                if (editorValue.DataTypeConfiguration is MediaPicker3Configuration configuration)
-                {
-                    dtos = PersistTempMedia(dtos, configuration);
-                }
+                return base.FromEditor(editorValue, currentValue);
+            }
 
-                // Clean up redundant/default data
-                foreach (JObject? dto in dtos.Values<JObject>())
-                {
-                    MediaWithCropsDto.Prune(dto);
-                }
+            List<MediaWithCropsDto>? mediaWithCropsDtos = _jsonSerializer.Deserialize<List<MediaWithCropsDto>>(jsonArray.ToJsonString());
+            if (mediaWithCropsDtos == null)
```

Comment:
More pattern matching 😁 
```suggestion
            if (mediaWithCropsDtos is null)
```